freqtrade/strategy/strategy_helper.py

```diff
@@ -83,3 +97,94 @@ def stoploss_from_open(open_relative_stop: float, current_profit: float) -> floa
+                fmt: Optional[Union[str, Callable[[KwArg(str)], str]]] = None,
+                ffill: bool = True) -> Callable[[Callable[[Any, DataFrame, dict], DataFrame]],
+                                                Callable[[Any, DataFrame, dict], DataFrame]]:
+    """
+    A decorator for populate_indicators_Nn(self, dataframe, metadata), allowing these functions to
+    define informative indicators.
+
+    Example usage:
+
+        @informative('1h')
+        def populate_indicators_1h(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+            dataframe['rsi'] = ta.RSI(dataframe, timeperiod=14)
+            return dataframe
+
+    :param timeframe: Informative timeframe. Must always be higher than strategy timeframe.
+    :param asset: Informative asset, for example BTC, BTC/USDT, ETH/BTC. Do not specify to use
+    current pair.
+    :param fmt: Column format (str) or column formatter (callable(name, asset, timeframe)). When not
+    specified, defaults to {asset}_{name}_{timeframe} if asset is specified, or {name}_{timeframe}
+    otherwise.
+    * {asset}: name of informative asset, provided in lower-case, with / replaced with _. Stake
+    currency is not included in this string.
+    * {name}: user-specified dataframe column name.
+    * {timeframe}: informative timeframe.
+    :param ffill: ffill dataframe after mering informative pair.
+    """
+    def decorator(fn: Callable[[Any, DataFrame, dict], DataFrame]):
+        @wraps(fn)
+        def wrapper(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+            nonlocal timeframe, asset, fmt
+            # Modifying variables inherited from parent scope poisons other wrapper instances!
+            this_asset = asset
+            this_fmt = fmt
+
+            # Default format.
+            if this_fmt is None:
+                this_fmt = '{name}_{timeframe}'
+                if this_asset:
+                    this_fmt = '{asset}_' + this_fmt
+
+            # Not specifying an asset will define informative dataframe for current pair.
+            if not this_asset:
+                this_asset = metadata['pair']
+            assert isinstance(this_asset, str)
+
+            # Not specifying quote currency will use stake currency.
+            if '/' not in this_asset:
+                this_asset = f'{asset}/{self.config["stake_currency"]}'
+
+            inf_metadata = {'pair': this_asset, 'timeframe': timeframe}
+            inf_dataframe = self.dp.get_pair_dataframe(this_asset, timeframe)
+            inf_dataframe = fn(self, inf_dataframe, inf_metadata)
+
+            # Clear stake currency from specified asset name.
+            fmt_asset = re.sub(rf'([^/])(/{self.config["stake_currency"]})?', r'\1',
+                               this_asset).lower()
+            # Non-stake quote currency will be kept in asset name, replace / separator with _.
+            fmt_asset = fmt_asset.replace('/', '_')
```

Comment:
i think this will be quite confusing.
if i specify `asset='BTC/USDT'` - why won't my new column be called `BTC/USDT_rsi_1h` (without any way for me to override this behaviour).

Having a `/` in the column name is perfectly fine (you just can't have it in a filename) - so i'd not enforce this onto users without real reason

Reply:
Reason is that stake currency is redundant information which just introduces extra typing without conveying much information. Likewise i replaced `/` with `_` for consistency. I understand we can have columns in any format, but there is no reason to opt in for inconsistent option if we can have a consistent one. Overriding is good though, so im making it so user can specify `fmt='{asset}_{name}_{timeframe}'` to get `BTC/USDT_rsi_1h` while `fmt='{asset_short}_{name}_{timeframe}'` would keep simplified `btc_rsi_1h`. Note that by default this does not exclude quote currency if it is different from stake currency, as that conveys useful information.

Reply:
it's not really redundant if i need 2 (or more) 1h pairs (with the same indicators) ... then the asset becomes quite important.

Reply:
I do not follow. Could you claraify what you mean by "2 (or more) 1h pairs (with the same indicators)"?

Reply:
you claim that stake-currency is redundant - but is it really ? 

if i use `ETH/USDT` and `ETH/BUSD` and `ETH/BTC` (all on 1h) - and i switch stake currency around - i'll get different outputs due to the "redundant" stuff.

i might get 
`ETH_1h`, `ETH_BUSD_1h` and `ETH_BTC_1h` ,
`ETH_USDT_1h`, `ETH_1h` and `ETH_BTC_1h` ,
`ETH_USDT_1h`, `ETH_BUST_1h` and `ETH_1h` 

just because i switched stake-currency. As such - i cannot rely on the results i get as the "magic" is creating inconsistent results.

Reply:
But this is far from common case. Vast majority will use `BTC/USDT` as informative pair. Thats it. Now when you switch stake currency you would have to update your strategy. This `BTC/{pair}` helper to avoid hardcoding stake currency goes out of the window. Your example is very valid, but as it is not common i would suggest to document how to handle such case instead of embedding stake currency all the time and making strategy stake-dependent.